heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -26,28 +33,46 @@
 
 import com.twitter.heron.api.generated.TopologyAPI;
 import com.twitter.heron.proto.system.PackingPlans;
+import com.twitter.heron.proto.system.PhysicalPlans;
 import com.twitter.heron.spi.common.Config;
 import com.twitter.heron.spi.packing.PackingPlan;
 import com.twitter.heron.spi.packing.PackingPlanProtoDeserializer;
 import com.twitter.heron.spi.scheduler.IScalable;
 import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.NetworkUtils;
 import com.twitter.heron.spi.utils.Runtime;
+import com.twitter.heron.spi.utils.TMasterUtils;
+import com.twitter.heron.spi.utils.TopologyUtils;
+
+import static com.twitter.heron.api.Config.TOPOLOGY_UPDATE_DEACTIVATE_WAIT_SECS;
+import static com.twitter.heron.api.Config.TOPOLOGY_UPDATE_REACTIVATE_WAIT_SECS;
 
 /**
  * Class that is able to update a topology. This includes changing the parallelism of
  * topology components
  */
-public class UpdateTopologyManager {
+public class UpdateTopologyManager implements Closeable {
   private static final Logger LOG = Logger.getLogger(UpdateTopologyManager.class.getName());
 
+  private Config config;
   private Config runtime;
   private Optional<IScalable> scalableScheduler;
   private PackingPlanProtoDeserializer deserializer;
+  private ScheduledThreadPoolExecutor reactivateExecutorService;
 
-  public UpdateTopologyManager(Config runtime, Optional<IScalable> scalableScheduler) {
+  public UpdateTopologyManager(Config config, Config runtime,
+                               Optional<IScalable> scalableScheduler) {
+    this.config = config;
     this.runtime = runtime;
     this.scalableScheduler = scalableScheduler;
     this.deserializer = new PackingPlanProtoDeserializer();
+    this.reactivateExecutorService = new ScheduledThreadPoolExecutor(1);
+    this.reactivateExecutorService.setMaximumPoolSize(1);
+  }
+
+  @Override
+  public void close() {
+    this.reactivateExecutorService.shutdown();
```

Comment:
Cood catch, fixed. I changed to `shutdown()` when I was trying to get the CLI to not shutdown and kill the scheduler work before the reactivation finished. Instead I just waited on the future with `future.get(timeout)`.